apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -322,10 +355,49 @@ def mlset(self, keys, value):
         container, key = find_ml(self._data, keys)
         container[key] = value
 
+    def mlset_cast(self, keys, value):
+        type_msg = "The type of {} should be {}."
+
+        if isinstance(keys, str):
+            key_list = keys.split(".")
+        elif not isinstance(keys, list):
+            key_list = list(keys)
+
+        schema_keys = list(chain.from_iterable(zip(['properties'] * len(key_list), key_list)))
+
+        container, key = find_ml(self.validator.schema, schema_keys)
+
+        if key in container:
+            container = container[key]
+            if "type" in container:
+                val_type = container["type"]
+                if val_type == "integer":
+                    if value.isdigit():
```

Comment:
Hmm. Noticed that `-123` doesn't pass this. I think we can just call `int(value)` and if ValueError is raised, we just propagate that up.

If type casting is wrapped to try-except, we could add the expected value type to the exception:

```
try:
  value = int(value)
except ValueError as e:
  e.value_type = val_type
  raise e
```

This way we can move the UI error message to cli.py file.

Reply:
what's the point of `e.value_type = val_type`? doesn't change the error message at all.

Reply:
an unedited ValueError isn't very useful for the user either, it doesn't really tell where the problem is

Reply:
...missed the last sentence. I'm very good at reading these comments :')